### PR TITLE
Ports pools from Baystation12

### PR DESCRIPTION
**Pull request by quardbreak**
## Description of changes

This PR ports pool tiles, faucets and bath drain.

## Why and what will this PR improve

Allow to contain water on tiles. Also, it's fun!

## Authorship

@CrimsonShrike
https://github.com/Baystation12/Baystation12/pull/29425
https://github.com/Baystation12/Baystation12/pull/29550

## Changelog

:cl: CrimsonShrike and quardbreak
rscadd: Ported pool tiles and related to pool objects. You can build pool and keep water inside it.
/:cl:

### CODE REVIEW

File: code/game/turfs/flooring/flooring_premade.dm [hunk 1]
```diff
@@ -375,3 +375,14 @@
...
+
+//Water go splish
+/turf/simulated/floor/pool
+	name = "pool floor"
+	icon = 'icons/turf/flooring/pool.dmi'
+	icon_state = "pool"
+	initial_flooring = /decl/flooring/pool
+
+/turf/simulated/floor/pool/get_physical_height()
+	return -FLUID_OVER_MOB_HEAD
```
Comment (CrimsonShrike) [inline code/game/turfs/flooring/flooring_premade.dm]: See above

File: code/game/turfs/flooring/flooring.dm [hunk 2]
```diff
@@ -393,3 +395,16 @@
...
+	desc = "Sunken flooring designed to hold liquids."
+	icon = 'icons/turf/flooring/pool.dmi'
+	icon_base = "pool"
+	build_type = /obj/item/stack/tile/pool
+	flags = TURF_HAS_CORNERS | TURF_HAS_INNER_CORNERS | TURF_REMOVE_CROWBAR
+	footstep_type = /decl/footsteps/tiles
+	floor_smooth = SMOOTH_NONE
+	wall_smooth = SMOOTH_NONE
+	space_smooth = SMOOTH_NONE
+	height = -FLUID_OVER_MOB_HEAD * 2
```
Comment (CrimsonShrike) [inline code/game/turfs/flooring/flooring.dm:410]: height var and value returned by the preset are different.

File: code/game/objects/structures/drain.dm [hunk 3]
```diff
@@ -45,11 +50,48 @@
...
+	var/constructed_type = /obj/structure/hygiene/drain
+
 /obj/item/drain/attackby(var/obj/item/thing, var/mob/user)
 	if(isWrench(thing))
-		new /obj/structure/hygiene/drain(src.loc)
-		playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
-		to_chat(user, "<span class='warning'>[user] wrenches the [src] down.</span>")
+		new constructed_type(get_turf(src))
+		playsound(src, 'sound/items/Ratchet.ogg', 50, 1)
+		to_chat(user, SPAN_WARNING("\The [user] wrenches the [src] down."))
```
Comment (MistakeNot4892) [inline code/game/objects/structures/drain.dm]: `SPAN_NOTICE`

File: code/game/turfs/flooring/flooring.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/game/turfs/flooring/flooring.dm:410]: Preset no longer returns a distinct value, uses flooring instead.

File: code/modules/materials/recipes_stacks.dm [hunk 4]
```diff
@@ -91,4 +91,9 @@
 
 /datum/stack_recipe/tile/walnut
 	title = "walnut floor tile"
-	result_type = /obj/item/stack/tile/walnut
\ No newline at end of file
+	result_type = /obj/item/stack/tile/walnut
+
+
+/datum/stack_recipe/tile/metal/pool
+	title = "pool floot tile"
```
Comment (MistakeNot4892) [inline code/modules/materials/recipes_stacks.dm]: `pool floot`

File: code/game/objects/structures/drain.dm [hunk 5]
```diff
@@ -45,11 +50,48 @@
...
+	icon_state = "drain_bath"
+
+	var/closed = FALSE
+
+/obj/structure/hygiene/drain/bath/attack_hand(mob/user)
+	. = ..()
+	if(!welded)
+		closed = !closed
+		user.visible_message(SPAN_NOTICE("\The [user] has [closed ? "closed" : "opened"] the drain."))
+	update_icon()
```
Comment (MistakeNot4892) [inline code/game/objects/structures/drain.dm:76]: Should `return TRUE` to attack_hand here and below.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**